app/components/admin/participants/details/details.html+ect_pending.erb

```diff
@@ -0,0 +1,30 @@
+<h2 class="govuk-heading-m">Details</h2>
```

Comment:
Also, we have `ect_pending`, `mentor_pending`, `npq`, `npq_pending`. Are we missing something? :P

Reply:
Unfortunately, this convention is pretty out of our hands - it is how `versions` of the ViewComponent works - during compilation phase they search for files matching the name of the component (either in the same folder or in the folder with same name) and then they split it on dots, take the second part (normally `html` or `text`), split that on `+` sign and whatever follows is added to `context_lookup.versions` array. What's even more annoying is that they use that name to define public method `call_#{version}` so you're not allowed to use just any characters in the version...

I've spent quite a while searching for a more explicit way to pass/override template to be used for the component, the close second place was having a separate component per each type and one component that selects which one to render...

As per missing templates - yes, we're missing `ect` and `mentor`. As for now, all the ect and mentor profiles are pending validation - the exact requirement for ECT validations are quite different from NPQ, and since NPQ is a prioirty for now I have not added the other validations yet. So, for `ect` and `mentor` profiles, we'll be always displaying `ect_pending` and `mentor_pending` for now.

Reply:
Uhh, the magic and limitations surrounding `versions` sounds like a very good idea to have one component per user type and an extra component which uses a switch to display the right one. 

Saves anyone else the need to dive deep into ViewComponent versions and being horrified at what they discover :P

Reply:
And with your component test setup, we could easily unit test that the right component gets rendered for the right user. Which I don't think is particularly easy to verify with `versions` (or am I wrong and it is very easy?)

Reply:
It is easy enough to test that the correct content is displayed with standard content matchers. The one annoying thing is the sheer number of files in the director - 12 files just to handle that switch without any difference in behaviour...

Reply:
That being said, I think I'd like to bite the bullet here as I'd like this merged sooner than later. I am ahppy to refactor in next PR, but amount of conflicts arising here every day is really getting in the way...

Reply:
Agree with @kicferk1 I would be much happier seeing some extra components rather than being forced into this

Reply:
Ok, this is now changed to use mutliple sub classes instead of variants.